packages/matter.js/src/protocol/interaction/SubscriptionHandler.ts

```diff
@@ -713,7 +713,26 @@ export class SubscriptionHandler {
         );
     }
 
-    async attributeChangeListener<T>(path: AttributePath, schema: TlvSchema<T>, version: number, value: T) {
+    attributeChangeListener<T>(path: AttributePath, schema: TlvSchema<T>, version: number, value: T) {
+        const changeResult = this.attributeChangeHandler(path, schema, version, value);
+        if (MaybePromise.is(changeResult)) {
+            this.attributeUpdatePromises.push(
+                changeResult.then(() => {
+                    const promiseIndex = this.attributeUpdatePromises.indexOf(changeResult);
+                    if (promiseIndex !== -1) {
+                        this.attributeUpdatePromises.splice(promiseIndex, 1);
+                    }
```

Comment:
If you use a set for attributeUpdatePromises then this could just be `this.attributeUpdatePromises.delete`